lib/plenario2/core/actions/data_set_field_actions.ex

```diff
@@ -0,0 +1,29 @@
+defmodule Plenario2.Core.Actions.DataSetFieldActions do
+  import Ecto.Query
+  alias Plenario2.Core.Changesets.DataSetFieldChangesets
+  alias Plenario2.Core.Schemas.DataSetField
+  alias Plenario2.Repo
+
+  def create(meta_id, name, type, opts \\ "default null") do
+    params = %{
+      meta_id: meta_id,
+      name: name,
+      type: type,
+      opts: opts
+    }
+
+    DataSetFieldChangesets.create(%DataSetField{}, params)
+    |> Repo.insert()
+  end
+
+  def list_for_meta(meta) do
+    Repo.all(from f in DataSetField, where: f.meta_id == ^meta.id)
+  end
```

Comment:
The could be `Repo.get_by(DataSetField, meta_id: meta.id)` for more readability. Then you could rename the function to `list_by_meta/1` to be consistent with the API provided by `Repo`. There's also a `get_by!/3` if you want behavior that crashes for no results.

Reply:
I'm avoiding the `!` functions -- I like that if there's a problem it'll return `{:error, changeset information}` so that application controllers can pluck the error messages and format them for a response.

Reply:
Also, `get_by/3` expects a length of 1 result -- this is a listed result so it isn't going to work.

```
  1) test list fields for meta (DataSetFieldTests)
     test/plenario2/core/data_set_field_test.exs:25
     ** (Ecto.MultipleResultsError) expected at most one result but got 2 in query:
     
     from d in Plenario2.Core.Schemas.DataSetField,
       where: d.meta_id == ^697
     
     code: fields = DataSetFieldActions.list_for_meta(context.meta)
     stacktrace:
       (ecto) lib/ecto/repo/queryable.ex:73: Ecto.Repo.Queryable.one/4
       test/plenario2/core/data_set_field_test.exs:29: (test)
```